AU2/frontends/inquirer_cli.py

```diff
@@ -429,6 +445,104 @@ def render(html_component, dependency_context={}):
+                    new_values.append(p_data)
+                    values.append(p_data)
+            elif c == -1: # case where "*CONTINUE*" selected
+                break
+            else:
+                v = values[c]
+                q = [inquirer.Text(
+                    name="editpseudonym",
+                    message="Enter replacement",
+                    default=v.text
+                )]
+                try:
+                    p = inquirer_prompt_with_abort(q)["editpseudonym"]
+                except KeyboardInterrupt:
+                    continue
+                if p:
+                    # whitespace values => delete
+                    # could change this to some other input e.g. "-"
+                    if p.strip() == "":
+                        if c == 0: # if we don't catch this case then delete_pseudonym will throw an error down the line...
+                            print("Can't delete initial pseudonym!")
+                            continue
+
+                        q = [inquirer.Confirm(
+                            name="deletepseudonym",
+                            message=f"Do you wish to delete the pseudonym {v.text}?",
+                            default=False
+                        )]
+                        try:
+                            if inquirer_prompt_with_abort(q)["deletepseudonym"]:
+                                deleted_indices.add(c)
+                                values[c] = None
+                        finally:
+                            continue
+
+                    if c == 0: # intial pseudonym should always be valid forever
+                        valid_from = v.valid_from
+                    else:
+                        q = [inquirer.Text(
+                            name="dt",
+                            message=f"Enter start of validity (YYYY-MM-DD HH:MM)",
+                            default=v.valid_from.strftime(DATETIME_FORMAT) if v.valid_from else "",
+                            validate=optional_datetime_validator,
+                        )]
+                        try:
+                            dt_str = inquirer_prompt_with_abort(q)["dt"]
+                        except KeyboardInterrupt:
+                            continue
+                        valid_from = (None if dt_str == ""
+                                      else datetime.datetime.strptime(dt_str, DATETIME_FORMAT).astimezone(TIMEZONE))
+                    p_data = PseudonymData(p, valid_from)
+                    values[c] = p_data
+                    if c < old_n_values:
+                        edited[c] = p_data
+                    else:
+                        new_values[c-old_n_values] = p_data
+
+        return {html_component.identifier:
+                    ListUpdates(edited, new_values, deleted_indices)}
+
     elif isinstance(html_component, ArbitraryList):
         q = [inquirer.Checkbox(
             name=html_component.identifier,

```

Comment:
Please can you delete `ArbitraryList` and all references to it given your refactor into `EditablePseudonymList`? `ArbitraryList` is very badly named but appears to only apply to pseudonyms (it looks like it was meant to be for a list for which things can be added arbitrarily, but wasn't generalized due to time constraints in MW24 development)